.github/workflows/CI.yaml

```diff
@@ -0,0 +1,37 @@
+name: SonarCloud
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  build:
+    name: Build and analyze
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: 'zulu' # Alternative distribution options are available
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v4
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Cache Gradle packages
+        uses: actions/cache@v4
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Build and analyze
```

Comment:
- name: Set Gradlew permissions
        run: chmod +x gradlew

Reply:
There is "All checks have failed" and when we click on details, there is this error message : 
Run ./gradlew build sonar --info
/home/runner/work/_temp/c3644348-b754-42e3-ba8e-021b2b[9](https://github.com/BrewR-EPFL/BrewR/actions/runs/11212842619/job/31164749183?pr=26#step:6:10)865ee.sh: line 1: ./gradlew: Permission denied
Error: Process completed with exit code 126.

Reply:
So when you run/build those codes, it built in your computer ? 
I'm afraid to approve this PR and merging the one with errors.

Reply:
https://sonarcloud.io/summary/new_code?id=BrewR-EPFL_BrewR&pullRequest=26